site/docs/v1/tech/oauth/url-validation.adoc

```diff
@@ -0,0 +1,274 @@
+ * The wildcard can be a full or partial replacement of the path segment
+
+The following table provides some examples of valid and invalid wildcard patterns in the path.
+
+[cols="45m,10,45"]
+.Path Wildcards
+|===
+| Example | Valid | Reason
+
+|\https://example.com/path/*/resource
+|✓
+|&nbsp;
+
+|\https://example.com/p*/to/resource
+|✓
+|&nbsp;
+
+|\https://example.com/pass:[*]/par*tial/pass:[*]
+|✓
+|&nbsp;
+
+|\https://example.com/path/pass:[*]midpass:[*]/resource
+|✕
+|The path segment contains multiple wildcards
+
+|===
+
+=== Query String Values
+
+Wildcards are allowed in query string values with the following restrictions:
+
+ * Partial wildcards are not allowed for query string values
+
+Wildcards are _not_ allowed in query string names. The following table provides some examples of valid and invalid wildcard patterns in the query string.
+
+[cols="45m,10,45"]
+.Query String Wildcards
+|===
+| Example | Valid | Reason
+
+|\https://example.com?foo=*
+|✓
+|&nbsp;
+
+|\https://example.com?foo=pass:[*]&bar=pass:[*]&baz=blah
+|✓
+|&nbsp;
+
+|\https://example.com?foo=par*tial
+|✕
+|Partial wildcard replacement is not allowed for query string values
+
+|\https://example.com?*=blah
+|✕
+|Wildcards are not allowed in query string names
+
+|===
+
+== Wildcard Replacement Patterns
+
+The position where wildcards are allowed in configured values is just one half of the puzzle. Wildcards in each portion of the URL have different rules for the replacement values. Please note that allowed replacement values may not produce a valid URL. This section provides details on the allowed replacements for wildcards in each portion of the URL. A wildcard in the configured value must be replaced with at least one character in the concrete value.
+
+=== Domain
+
+Replacements for wildcards in the domain portion of the URL must not contain `.`, `:`, `/`, or `?` characters. The following table lists examples of valid and invalid replacements for valid wildcard patterns.
+
+[cols="25m,25m,10,40"]
+.Domain wildcard replacement
+|===
+| Pattern | Value | Valid | Reason
+
+|\https://*.example.com
+|\https://login.example.com
+|✓
+|&nbsp;
+
+|\https://auth*.example.com
+|\https://auth2.example.com
+|✓
+|&nbsp;
+
+|\https://auth*.example.com
+|\https://auth.example.com
+|✕
+|The value does not contain a character to replace the `*`
+
+|\https://*.example.com
+|\https://auth.customer.example.com
+|✕
+|The replacement contains a `.` character
+
+|===
+
+=== Port
+
+Replacement values for wildcards in the port portion of the URL must consist of one or more digits.
+
+[cols="25m,25m,10,40"]
+.Port wildcard replacement
+|===
+| Pattern | Value | Valid | Reason
+
+|\https://example.com:*
+|\https://example.com:2012
+|✓
+|&nbsp;
+
+|\https://example.com:*
+|\https://example.com:80b
+|✕
+|The replacement value contains a non-numeric character
+
+|===
+
+=== Path Segments
+
+Replacement values for wildcards in a path segment of the URL must not contain `/` or `?` characters.
```

Comment:
Do we test for URL escaped values? `%2F`?

Reply:
As I'm reading the docs, I was wondering if `https://example.com/path/*/resource` would match `https://example.com/path%2fto%2Fthe/resource` . I just tested on sandbox and it fails. But would probably be a good test to have.

Reply:
I think that fails because it doesn't have `/path/`. Pretty sure that `/path/to%2Fthe/resource` would pass.

I have a task to add some tests for the following:
* All of the examples in doc (valid and invalid)
* The [example](https://inversoft.slack.com/archives/C0407G6V5/p1678138231518359) you called out in Slack
* A couple of these escape sequences